src/main/java/com/imsweb/datagenerator/naaccr/NaaccrDataGenerator.java

```diff
@@ -270,16 +274,23 @@ public NaaccrDataGeneratorRule getTumorRule(String ruleId) {
         if (options == null)
             options = new NaaccrDataGeneratorOptions();
 
+        // Context contains information that is shared amongst the rules. Contains values we want to use for this patient.
+        Map<String, String> context = generateInitialPatientContext(numTumors);
+
         List<Map<String, String>> tumors = new ArrayList<>();
 
         // execute the patient rules once; we will copy the resulting values in each tumor
         Map<String, String> patient = new HashMap<>();
         for (NaaccrDataGeneratorRule rule : _patientRules)
             if (allPropertiesHaveValue(patient, rule.getRequiredProperties()))
-                rule.execute(patient, null, options);
+                rule.execute(patient, null, options, context);
 
         // create each tumor and add it to the return list
         for (int i = 0; i < numTumors; i++) {
+            if (context != null)
+                if (context.get("totalTumorCount") != null)
```

Comment:
You should use constants (public static final String) for any key you add to the context.  If they are used only in this class, it's OK to make them private, either way.

Reply:
Fixed.